extensions/cloudcredentials/aws/create.go

```diff
@@ -1,29 +1,53 @@
 package aws
 
 import (
+	"github.com/rancher/norman/types"
 	"github.com/rancher/shepherd/clients/rancher"
-	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
 	"github.com/rancher/shepherd/extensions/cloudcredentials"
-	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/defaults/stevetypes"
+	"github.com/rancher/shepherd/extensions/steve"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const awsCloudCredNameBase = "awsCloudCredential"
+const (
+	awsProvider         = "aws"
+	credentialNamespace = "cattle-global-data"
+)
 
-// CreateAWSCloudCredentials is a helper function that takes the rancher Client as a parameter and creates
-// an AWS cloud credential, and returns the CloudCredential response
-func CreateAWSCloudCredentials(rancherClient *rancher.Client) (*cloudcredentials.CloudCredential, error) {
-	var amazonEC2CredentialConfig cloudcredentials.AmazonEC2CredentialConfig
-	config.LoadConfig(cloudcredentials.AmazonEC2CredentialConfigurationFileKey, &amazonEC2CredentialConfig)
+// CreateAWSCloudCredentials is a helper function that creates V1 cloud credentials and waits for them to become active.
+func CreateAWSCloudCredentials(client *rancher.Client, credentials cloudcredentials.CloudCredential) (*v1.SteveAPIObject, error) {
+	users, err := client.Management.User.ListAll(&types.ListOpts{})
+	if err != nil {
+		return nil, err
+	}
 
-	cloudCredential := cloudcredentials.CloudCredential{
-		Name:                      awsCloudCredNameBase,
-		AmazonEC2CredentialConfig: &amazonEC2CredentialConfig,
+	secretName := namegenerator.AppendRandomString(awsProvider)
+	spec := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "cc-",
+			Namespace:    credentialNamespace,
+			Annotations: map[string]string{
+				"provisioning.cattle.io/driver": awsProvider,
+				"field.cattle.io/name":          secretName,
+				"field.cattle.io/creatorId":     users.Data[0].ID,
```

Comment:
is the 0th user always the one you expect? In the past, we've run into trouble with how norman sorts things

Reply:
It was in the case of admin/standard clients. That being said I do wonder if there is a better way to pull this ID without

Reply:
I think we should pass the user ID instead. How do we guarantee that the first element of the user's list is the user that created the credential?